ts/src/base/Exchange.ts

```diff
@@ -2802,8 +2802,8 @@ export default class Exchange {
         const extendsStr: Str = this.safeString (featuresObj, 'extends');
         if (extendsStr !== undefined) {
             featuresObj = this.omit (featuresObj, 'extends');
-            const extendObj = initialFeatures[extendsStr];
-            featuresObj = this.extend (extendObj, featuresObj); // Warning, do not use deepExtend here, because we override only one level
+            const extendObj = this.featuresMapper (initialFeatures, extendsStr);
+            featuresObj = this.deepExtend (extendObj, featuresObj);
```

Comment:
this is updated/corrected way to support recursive extension (eg. `swap extends default2, default2 extends default1`)